spec/locale.html

```diff
@@ -452,5 +550,265 @@ <h1>
+        1. Assert: _region_ is not *undefined*.
+        1. Let _list_ be a List of unique canonical time zone identifiers, which must be String values indicating a canonical Zone name of the IANA Time Zone Database, of those in common use in _region_. The list is empty if no time zones are commonly used in _region_. The list is sorted according to lexicographic code unit order.
+        1. Return CreateArrayFromList( _list_ ).
+      </emu-alg>
+    </emu-clause>
+
+    <emu-clause id="sec-locale-direction" type="abstract operation">
+      <h1>
+        LocaleDirection (
+          _loc_: an Intl.Locale,
+        ): a String or *undefined*
+      </h1>
+      <dl class="header">
+        <dt>description</dt>
+        <dd>The following algorithm refers to Locale data specified in <a href="https://www.unicode.org/reports/tr35/tr35-general.html#Layout_Elements">UTS 35's Layouts Elements</a>.</dd>
+      </dl>
+      <emu-alg>
+        1. Let _locale_ be _loc_.[[Locale]].
+        1. Let _script_ be GetLocaleScript(_locale_).
+        1. If _script_ is *undefined*, then
+          1. Let _maximal_ be the result of the <a href="https://unicode.org/reports/tr35/#Likely_Subtags">Add Likely Subtags</a> algorithm applied to _locale_. If an error is signaled, return *undefined*.
+          1. Set _script_ to GetLocaleScript(_maximal_).
+          1. If _script_ is *undefined*, return *undefined*.
+        1. If the default general ordering of characters within a line in _script_ is right-to-left, return "rtl".
+        1. If the default general ordering of characters within a line in _script_ is left-to-right, return "ltr".
+        1. Return *undefined*.
+      </emu-alg>
+      <emu-note>
+        When the direction of default general ordering of characters within a line in the _script_ cannot be determined, or the direction is neither right-to-left nor left-to-right, then *undefined* will be returned.
+      </emu-note>
+      <emu-note>
+        Text in a language may have more than one possible direction. Text in a language written in one script may also have more than one possible direction. Therefore, if there is direction information included in the metadata associated with a piece of text, that direction information should be considered as the primary source of truth. The information returned by LocaleDirection should only be used as a fallback mechanism while there is no metadata associated with the text to provide the direction information of the text. It should only be used while the direction information is not present but language information is known. Web developers should not rely too much on this API to decide the direction of the text.
+      </emu-note>
+    </emu-clause>
+
+    <emu-clause id="sec-weekday-to-string" type="abstract operation">
+      <h1>
+        WeekdayToString (
+          _fw_: a String,
+        ): a String
+      </h1>
+      <dl class="header">
+      </dl>
```

Comment:
Please rename for specificity and add a description, e.g.
```suggestion
      <h1>
        FirstDayOfWeekToUValue (
          _fw_: a String,
        ): a String
      </h1>
      <dl class="header">
        <dt>description</dt>
        <dd>It converts a string that numerically represents a day of the week to a valid <a href="https://unicode.org/reports/tr35/#UnicodeFirstDayIdentifier">Unicode First Day Identifier</a> defined in <a href="https://unicode.org/reports/tr35/#Key_And_Type_Definitions_">Unicode Technical Standard #35 Part 1 Core Section 3.6.1 Key and Type Definitions</a>, and returns any other string unmodified.</dd>
      </dl>
```